contracts/src/PermissionedStakeTable.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { BN254 } from "bn254/BN254.sol";
+import { EdOnBN254 } from "./libraries/EdOnBn254.sol";
+
+/**
+ * @title SimpleStakeTable
+ * @dev An stake table mapping with owner-only access control.
+ */
+contract PermissionedStakeTable is Ownable {
+    event Added(NodeInfo[]);
+    event Removed(NodeInfo[]);
+
+    error StakerAlreadyExists(BN254.G2Point);
+    error StakerNotFound(BN254.G2Point);
+
+    struct NodeInfo {
+        /// The consensus signing key
+        BN254.G2Point blsVK;
+        /// The consensus signing key. Only used for storage in this contract.
+        EdOnBN254.EdOnBN254Point schnorrVK;
+        /// Is the Node DA Node? Only used for storage in this contract.
+        bool isDA;
+    }
+
+    // State mapping from staker IDs to their staking status
+    mapping(bytes32 nodeID => bool isStaker) private stakers;
+
+    constructor(address initialOwner, NodeInfo[] memory initialStakers) Ownable(initialOwner) {
```

Comment:
in what scenarios do we imagine the initialOwner being different from the deployer?

Reply:
I was thinking if we want to set the multisig as owner during deployment. But yes, we should probably just do the same as with our other contracts.

Reply:
yes that's valid but agree to keep it inline with other deployments. Confirmed with Mat and Tobias that there's no need for a different initialOwner other than the deployer since we immediately change ownership to the multisig.

Reply:
So I think I fixed this